Improve ET_ASSERT_UNREACHABLE_MSG to use formatting

dulinriley opened the pull request:
Summary:
The macro `ET_ASSERT_UNREACHABLE_MSG` didn't allow for format
strings unlike the other `ET_CHECK` macros.
Add this capability and fix any users of it.

Reviewed By: dbort

Differential Revision: D65693396

dbort reviewed (approved):
Please fix the lint job before landing

dulinriley committed:
Improve ET_ASSERT_UNREACHABLE_MSG to use formatting (#6802)

Summary:

The macro `ET_ASSERT_UNREACHABLE_MSG` didn't allow for format
strings unlike the other `ET_CHECK` macros.
Add this capability and fix any users of it.

Reviewed By: dbort

Differential Revision: D65693396
